eva-lib/src/main/java/uk/ac/ebi/eva/lib/repository/VariantEntityRepositoryImpl.java

```diff
@@ -95,7 +97,12 @@ public Long countByRegionsAndComplexFilters(List<Region> regions, List<VariantEn
     private List<VariantEntity> findByComplexFiltersHelper(Query query, List<VariantEntityRepositoryFilter> filters,
                                                            List<String> exclude, Pageable pageable) {
 
-        applyFilters(query, filters);
+        if (filters.size() > 0){
+            List<Criteria> criteriaList = getFiltersCriteria(filters);
+            for (Criteria criteria : criteriaList) {
+                query.addCriteria(criteria);
+            }
+        }
```

Comment:
This block is still part of building the query. Could this be encapsulated in another method?